apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -360,36 +377,48 @@ export const getResourcePermalinkTree = async (
   siteId: number,
   resourceId: number,
 ) => {
-  const resourcePermalinks = await db
-    .withRecursive("Ancestors", (eb) =>
-      eb
-        // Base case: Get the actual resource
-        .selectFrom("Resource")
-        .where("Resource.siteId", "=", siteId)
-        .where("Resource.id", "=", String(resourceId))
-        .select(defaultResourceSelect)
-        .unionAll((fb) =>
-          fb
-            // Recursive case: Get all the ancestors of the resource
-            .selectFrom("Resource")
-            .where("Resource.siteId", "=", siteId)
-            .innerJoin("Ancestors", "Ancestors.parentId", "Resource.id")
-            .select(defaultResourceSelect),
-        ),
-    )
-    .selectFrom("Ancestors")
-    .select("Ancestors.permalink")
-    .execute()
+  return db.transaction().execute(async (tx) => {
+    // Guard against invalid resource
+    const resource = await getById(tx, {
+      siteId,
+      resourceId,
+    }).executeTakeFirst()
+    if (!resource) {
+      return []
+    }
```

Comment:
how would we distinguish between a resource being empty and a resource not existing in this case?